src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts

```diff
@@ -93,11 +93,19 @@ export class AssessmentHeaderComponent {
         // if there is no 'save' button
         if (this.result?.completionDate) {
             return true;
+        } else if (Result.hasNonEmptyAssessmentNote(this.result)) {
+            // this is almost identical to submitDisabled, but without the assessmentsAreValid check
+            // otherwise, we wouldn't be able to save the assessment note without making prior changes to the feedback
+            return !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
         } else {
-            return !this.assessmentsAreValid || !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
+            return this.saveDisabledWhenNoAssessmentNotePresent;
         }
     }
 
+    get saveDisabledWhenNoAssessmentNotePresent() {
+        return !this.assessmentsAreValid || !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
+    }
+
     get submitDisabled() {
         return !this.assessmentsAreValid || !this.isAssessor || this.saveBusy || this.submitBusy || this.cancelBusy;
     }
```

Comment:
These two methods are the same, or am I missing something?

Reply:
Before this PR the save and submit logics where quite intermixed which lead to consistency issues across how different exercise types were handled during assessment. We would like to keep them distinct now to avoid that. Therefore, I think having this single-line duplication is fine.